### ISSUE TITLE
luxon integration for pickers

### ISSUE THREAD

**Issue by hihuz** [ant-design/ant-design#29710]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Hello :wave: First of all thanks a lot for your amazing work on antd!

antd pickers have existing integrations for dayjs and date-fns (and the current main integration moment), but not for [luxon](https://github.com/moment/luxon).

I can see that there was some interest in such an integration in [this issue](https://github.com/ant-design/ant-design/issues/22732) which is now closed.

Would you still be interested for volunteers to implement such an integration?

We are using antd & luxon in our project at my company and we have recently written our own picker integration (generate config) for luxon, so we could open a pull request to suggest our implementation.

Let me know what you think.

Thank you!

### What does the proposed API look like?
Basically there wouldn't be any new API, the luxon integration would implement the same `GenerateConfig` interface as the other integrations.

One notable difference with luxon is that it strictly uses the [standard Intl API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl) for localisation, which does not have any notion of localised weeks for now.

Therefore there would be some differences with other pickers regarding the handling of weeks, but we can discuss this further in the upcoming pull request if you are interested.

```typescript
import { DateTime, Info } from 'luxon';

const generateConfig: GenerateConfig<DateTime> = {
  ...
}
```

**Comment by hihuz** [comment]
The pull request was created on the picker repo: https://github.com/react-component/picker/pull/230

Any time spent reviewing it is greatly appreciated, and I'll create another PR against the main repo to document the feature when the discussions and feedback have been handled in the implementation PR.

Thanks a lot!

**Comment by yoyo837** [comment]
Since react-component/picker#230 was merged, there should be some other work in `antd` that needs to be continued, thanks for your great works.

**Comment by hihuz** [comment]
Thank you @yoyo837 🙏 

As far as I can tell the changes necessary to the `ant-design` repo are only:
* bump rc-picker version
* add documentation for luxon usage

Do I have that correct?

I have created a draft pull request here for these changes: https://github.com/ant-design/ant-design/pull/41580

The reason it is still a draft is that I unfortunately do not speak chinese 😞 I could try to auto generate a translation but that would probably not be very good. 

Do you know if anybody could assist me with translating the documentation changes as well as the changelog entry?

Thank you very much 🙏

**Comment by yoyo837** [comment]
> 

Yes. We can translate your English documents to Chinese.

**Comment by yoyo837** [comment]
Close by #41580
